crates/core/src/models/mod.rs

```diff
@@ -234,3 +239,73 @@ impl ConvertibleStarknetBlock for BlockWithTxs {
         Ok(block.into())
     }
 }
+
+pub struct StarknetEvent(Event);
+
+impl StarknetEvent {
+    pub fn new(sn_event: Event) -> Self {
+        Self(sn_event)
+    }
+}
+
+#[async_trait]
+impl ConvertibleStarknetEvent for StarknetEvent {
+    async fn to_eth_log(&self, _client: &dyn KakarotClient) -> Result<Log, KakarotClientError> {
+        // let starknet_block_id = StarknetBlockId::Tag(BlockTag::Latest);
+        // let address = match client.get_evm_address(&self.0.from_address, &starknet_block_id).await {
+        //     Ok(address) => address,
+        //     Err(_) => {
+        //         return Err(KakarotClientError::OtherError(anyhow::anyhow!(
+        //             "Starknet event address has no EVM mapping",
+        //         )));
+        //     }
+        // };
+
+        // Convert the StarkNet address to a byte array
+        let sn_bytes = self.0.from_address.clone().to_bytes_be();
+        // Take the last 20 bytes of the StarkNet address to create the Ethereum address
+        let address = Address::from_slice(&sn_bytes[12..32]);
+
+        let topics: Result<Vec<H256>, KakarotClientError> = self
+            .0
+            .keys
+            .chunks(2)
+            .map(|chunk| {
+                let low = BigUint::from_bytes_be(&chunk[0].to_bytes_be());
+                let high = match chunk.get(1) {
+                    Some(h) => BigUint::from_bytes_be(&h.to_bytes_be()),
+                    None => {
+                        return Err(KakarotClientError::OtherError(anyhow::anyhow!(
+                            "Not an convertible event: High value doesn't exist",
+                        )));
+                    }
+                };
+                let result = low + (BigUint::from(2u128).pow(128u32) * high);
+                // Converts the result to bytes.
+                let bytes = result.to_bytes_be();
+                // If the length of bytes is less than 32, prepends it with zeros to make it 32 bytes long.
+                let bytes = once(0u8).cycle().take(32 - bytes.len()).chain(bytes.into_iter()).collect::<Vec<_>>();
+                Ok(H256::from_slice(&bytes))
+            })
+            .collect();
+
+        let topics = match topics {
+            Ok(topics) => topics,
+            Err(err) => return Err(err),
+        };
+
+        let data: Bytes = self.0.data.iter().flat_map(|felt| felt.to_bytes_be()).collect::<Vec<u8>>().into();
+
+        Ok(Log {
+            address,
+            topics,
+            data,
+            transaction_hash: Option::None,
```

Comment:
will we ever be able to retrieve this info?

are they provided in normal EVM calls?
say you get_logs on Alchemy, for a dummy polygon tx?

Reply:
i assume you can set this for each log at the level of the starknet transaction. we can take a transaction hash as a parameter and do that.

i will look into adding an optional argument to this so a transaction level operation can pass it in

Reply:
ok i see

Reply:
i can add each of the block/transaction params as option type'ed arguments to the interface/impl, or we can wait when we feel the demand for them and see what makes sense then.

will nap and see what your thoughts are.